### PR TITLE
Housekeeping / Review location for binary-related code for the random scheme

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#557

# Overview

- [x] I have relocated binary serializer / deserializer to the `http-bridge` package, in the `Binary` module, next to the others.

- [x] Doing so, I have also slightly reviewed the signatures to make them a bit more consistent overall and easier to compose in the compatibility module.

- [x] I have adjusted the compatibility module to use these new functions.

- [x] I have also moved (and adjusted) tests accordingly

# Comments

### CODE REVIEW

File: lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/RandomSpec.hs [hunk 1]
```diff
@@ -166,96 +107,30 @@ generateTest1 :: GenerateKeyFromSeed
 generateTest1 = GenerateKeyFromSeed
     { mnem = defMnemonic
     , pwd = pp ""
-    , rootKey = xprv16 "b84d0b6db447911a98a3ade98145c0e8323e106f07bc17a99c2104c2688bb7528310902a3cec7e262ded6a4369ec1f48966a6b48b1ee90aa00e61b95417949f81258854ab44b0cfda59bd68fbd87f280841a390068049df0f8a903c94ba65b7aa4762129a6c83acfda5b257eaeb73ec5fee1518b6674fdc7891fe23f06174421"
+    , rootKey = xprv16
+        "b84d0b6db447911a98a3ade98145c0e8323e106f07bc17a99c2104c2688bb752831090\
+        \2a3cec7e262ded6a4369ec1f48966a6b48b1ee90aa00e61b95417949f81258854ab44b\
+        \0cfda59bd68fbd87f280841a390068049df0f8a903c94ba65b7aa4762129a6c83acfda\
+        \5b257eaeb73ec5fee1518b6674fdc7891fe23f06174421"
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/RandomSpec.hs:114]: No change in the content, just wrapping for line-length.

File: lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/RandomSpec.hs [hunk 2]
```diff
@@ -166,96 +107,30 @@ generateTest1 :: GenerateKeyFromSeed
...
 generateTest2 :: GenerateKeyFromSeed
 generateTest2 = GenerateKeyFromSeed
     { mnem = defMnemonic
     , pwd = pp "4a87f05fe25a57c96ff5221863e61b91bcca566b853b616f55e5d2c18caa1a4c"
-    , rootKey = xprv16 "b842ae13cbb31b7d96910472bbed5c8729c764d66af81b48120a6a583eae55faf78c24765e0c9826f4d095f3e6addb4bda68df322b220d3c08b8a5b414232d101258854ab44b0cfda59bd68fbd87f280841a390068049df0f8a903c94ba65b7aa4762129a6c83acfda5b257eaeb73ec5fee1518b6674fdc7891fe23f06174421"
+    , rootKey = xprv16
+        "b842ae13cbb31b7d96910472bbed5c8729c764d66af81b48120a6a583eae55faf78c247\
+        \65e0c9826f4d095f3e6addb4bda68df322b220d3c08b8a5b414232d101258854ab44b0c\
+        \fda59bd68fbd87f280841a390068049df0f8a903c94ba65b7aa4762129a6c83acfda5b2\
+        \57eaeb73ec5fee1518b6674fdc7891fe23f06174421"
```
Comment (KtorZ) [inline lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/RandomSpec.hs:125]: No change in the content, just wrapping for line-length.

File: lib/http-bridge/src/Cardano/Wallet/HttpBridge/Compatibility.hs [hunk 3]
```diff
@@ -111,53 +109,35 @@ instance PersistTx (HttpBridge network) where
...
-
-emptyAttributes :: CBOR.Encoding
-emptyAttributes = CBOR.encodeMapLen 0
+    keyToAddress k = Address
+        $ CBOR.toStrictByteString
+        $ CBOR.encodeAddress (getRawKey k)
+            [ encodeDerivationPathAttr pwd acctIx addrIx ]
+      where
+        (acctIx, addrIx) = derivationPath k
+        pwd = payloadPassphrase k
```
Comment (KtorZ) [inline lib/http-bridge/src/Cardano/Wallet/HttpBridge/Compatibility.hs:140]: I've inlined and removed all the intermediate functions `keyToAddressWith`, `attributesWithProtocolMagic` etc ... In the end, I find it much clearer to have things inlined as above and I don't think the little duplication of rewriting `Address $ CBOR.toStrictByteString` is really harmful ..

File: lib/http-bridge/test/unit/Cardano/Wallet/HttpBridge/BinarySpec.hs [hunk 4]
```diff
@@ -143,6 +179,67 @@ spec = do
...
+
+    describe "Golden Tests for Byron Addresses w/ random scheme (Mainnet)" $ do
+        it "decodeDerivationPath - mainnet - initial account" $
+            decodeDerivationPathTest DecodeDerivationPath
+                { mnem =
+                    arbitraryMnemonic
+                , addr =
+                    "82d818584283581ca08bcb9e5e8cd30d5aea6d434c46abd8604fe4907d\
+                    \56b9730ca28ce5a101581e581c22e25f2464ec7295b556d86d0ec33bc1\
+                    \a681e7656da92dbc0582f5e4001a3abe2aa5"
```
Comment (KtorZ) [inline lib/http-bridge/test/unit/Cardano/Wallet/HttpBridge/BinarySpec.hs:193]: Note that these used to be base58-encoded for no particular reason. In the end, it was more confusing than anything else so I've re-encoded those as hexadecimal to not suggest that these are actual user-facing addresses.

**Review by paweljakubas — APPROVED** [review]
LGTM!

File: lib/http-bridge/test/unit/Cardano/Wallet/HttpBridge/BinarySpec.hs (hunk 4 above)
Comment (rvl) [reply lib/http-bridge/test/unit/Cardano/Wallet/HttpBridge/BinarySpec.hs:193]: There _were_ actual user-facing address. I generated this test data with the old wallet API and copy&pasted directly into the tests.

Comment (KtorZ) [reply lib/http-bridge/test/unit/Cardano/Wallet/HttpBridge/BinarySpec.hs:193]: Of course, what I meant is that the fact that they were in base58 was unnecessary for the test and confused the reader into thinking that addresses had to be encoded in base58 to encode / decode the derivation path.
In this stage in testing, we only care about the binary payload of the address and not much about the user facing encoding. Hence why I switched from base58 to hexadecimal.

**Pull request merged**